File: SDS.Clould/SDS.Clould.WebAPI/DataModels/FoundPackageDataModel.cs
namespace SDS.Clould.WebAPI.DataModels
{
    using System;
    using System.Linq.Expressions;

    using SDS.Clould.Models;

    public class FoundPackageDataModel
    {
        public static Expression<Func<Package, FoundPackageDataModel>> FromData
        {
            get
            {
                return p => new FoundPackageDataModel
                {
                    Id = p.Id,
                    Price = p.Price,
                    Space = p.Space,
                    Kilograms = p.Kilograms,
                    FromTown = p.FromTown,
                    ToTown = p.ToTown,
                    Deadline = p.Deadline
                };
            }
        }

        public int Id { get; set; }

        public decimal Price { get; set; }

        // in square meters
        public int Space { get; set; }

        public int Kilograms { get; set; }

        public string FromTown { get; set; }

        public string ToTown { get; set; }

        public DateTime Deadline { get; set; }
    }
}

File: SDS.Mobile/app/src/main/java/com/example/spawneddeliveryservice/models/FoundPackageDataModel.java
package com.example.spawneddeliveryservice.models;

import org.json.JSONObject;

import java.util.Date;

public class FoundPackageDataModel {
    public int id;
    public double price;
    // in square meters
    public int space;
    public int kilograms;
    public String fromTown;
    public String toTown;
    public Date deadline;

    public static FoundPackageDataModel FromModel(String json){
        try {
            JSONObject jObject = new JSONObject(json);
            FoundPackageDataModel returnedModel = new FoundPackageDataModel();

            returnedModel.id = jObject.getInt("Id");
            returnedModel.price = jObject.getDouble("Price");
            returnedModel.space = jObject.getInt("Space");
            returnedModel.kilograms = jObject.getInt("Kilograms");
            returnedModel.fromTown = jObject.getString("FromTown");
            returnedModel.toTown = jObject.getString("ToTown");
            returnedModel.deadline = ConstantsDataModels.getDateFormatter().parse(jObject.getString("Deadline"));

            return returnedModel;
        }
        catch (Exception e){
            return null;
        }
    }
}
